Bottom: 9bc2c2f8ee2a501715d268f94fa72c3c0bf867a9
Top:    6d48965af9aa80946db57f7f0f90dd75e19b1035
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-20 08:51:50 +0800

Refresh of vhost-part

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index e7d3fd3..6691b9d 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2481,6 +2481,7 @@ static int tun_sendmsg(struct socket *sock, struct msghdr *m, size_t total_len)
 	struct tun_file *tfile = container_of(sock, struct tun_file, socket);
 	struct tun_struct *tun = tun_get(tfile);
 	struct tun_msg_ctl *ctl = m->msg_control;
+	struct xdp_buff *xdp;
 
 	if (!tun)
 		return -EBADFD;
@@ -2492,11 +2493,7 @@ static int tun_sendmsg(struct socket *sock, struct msghdr *m, size_t total_len)
 		rcu_read_lock();
 
 		for (i = 0; i < n; i++) {
-			struct xdp_buff *x = (struct xdp_buff *)ctl->ptr;
-			struct xdp_buff *xdp = &x[i];
-
-			xdp_set_data_meta_invalid(xdp);
-			xdp->rxq = &tfile->xdp_rxq;
+			xdp = &((struct xdp_buff *)ctl->ptr)[i]);
 			tun_xdp_one(tun, tfile, xdp);
 		}
